core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/ComparisonIndexer.java

```diff
@@ -47,7 +47,6 @@ public ComparisonIndexer(JoinerType comparisonJoinerType,
 
     @Override
     public void put(IndexProperties indexProperties, Tuple_ tuple, Value_ value) {
-        Objects.requireNonNull(value);
```

Comment:
No, but all the indexers form a chain, at the bottom of which is a `NoneIndexer` - this check has been moved there, which means it will only ever be executed once. At the very end of the chain.